Snowflake.Data/Core/FastParser.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Snowflake.Data.Core
+{
+    public class FastParser
+    {
+
+        public static Int64 FastParseInt64(byte[] s, int offset, int len)
```

Comment:
What about adding _private static Int64 FastParseInt(byte[] s, int offset, int len, UInt64 maxValue)_, with the same code/logic than _FastParseInt64_ and _FastParseInt32_ and have them call FastParseInt with the correct max value? FastParseInt16 and FastParseByte could also be added easily and catch overflow ahead of FastParseInt32.

Reply:
There is a good reason for having separate implementations of FastParseInt32 and FastParseInt64. 
FastParseInt32 is faster because it only works with 32-bit integers. When the end user calls DataReader.GetInt(i) we know he only expects a 32-bit value so we can call the faster version FastParseInt32 directly.
I don't see a good reason to define a FastParseInt method.

Reply:
I am not sure how much this would affect the performance, but if this is the reason behind those almost 2 identical functions then it's fine.

Reply:
![image](https://user-images.githubusercontent.com/51655419/111706739-13b8da00-8843-11eb-89a6-23ed3521005b.png)

This is a benchmark of different parsing methods. As you can see FastParseInt32 is about 30-40% faster than FastParseInt64. I think it is well worth the effort to enable that speedup.

TypeConverter was the old code before this PR. As you can see, the old code was 57 times slower than the new version on .Net Framework